src/main/java/appeng/core/stats/IAdvancementTrigger.java

```diff
@@ -19,9 +19,10 @@
 package appeng.core.stats;
 
 
-public enum AchievementType
-{
+import net.minecraft.entity.player.EntityPlayerMP;
 
-	Craft, CraftItem, Pickup, Custom
 
+public interface IAdvancementTrigger
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Annotate the "IAdvancementTrigger" interface with the @FunctionalInterface annotation [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1609)